pkg/backporter/httpcache.go

```diff
@@ -0,0 +1,146 @@
+package backporter
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	xFromCache       = "X-From-Cache"
+	cacheRefreshFreq = 10 * time.Minute
+)
+
+type bugzillaCache struct {
+	lock  sync.Mutex
+	cache map[string][]byte
+}
+
+func (bc *bugzillaCache) get(key string) ([]byte, bool) {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	cachedVal, ok := bc.cache[key]
+	return cachedVal, ok
+}
+
+func (bc *bugzillaCache) set(key string, respBytes []byte) {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	bc.cache[key] = respBytes
+}
+
+func (bc *bugzillaCache) delete(key string) {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	delete(bc.cache, key)
+}
+func newBugzillaCache() *bugzillaCache {
+	return &bugzillaCache{cache: map[string][]byte{}}
+}
+
+// cachingTransport is an implementation http.RoundTripper
+// which first checks for cached values
+type cachingTransport struct {
+	cache     *bugzillaCache
+	transport http.RoundTripper
+}
+
+// RoundTrip will first check if there are any cached responses and return that
+// if not it will make an HTTP call using the DefaultTransport
+func (t *cachingTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if req.Method != "GET" {
+		return t.transport.RoundTrip(req)
+	}
+	var err error
+	var resp *http.Response
+	g := new(errgroup.Group)
+	g.Go(func() error {
+		resp, err = t.transport.RoundTrip(req)
+		if err != nil {
+			t.cache.delete(req.URL.String())
```

Comment:
Don't delete here, networks and services may have hiccups, no reason to delete cache entries